src/server/pps/server/poller.go

```diff
@@ -119,27 +147,133 @@ func (a *apiServer) pollPipelines(pachClient *client.APIClient) {
 		// always rm 'pipeline', to advance loop
 		delete(etcdPipelines, pipeline)
 
-		// generate an etcd event for 'pipeline' by reading it & writing it back
+		// generate a pipeline event for 'pipeline'
 		log.Debugf("PPS master: polling pipeline %q", pipeline)
-		var curPI pps.EtcdPipelineInfo
-		_, err := col.NewSTM(ctx, a.env.GetEtcdClient(), func(stm col.STM) error {
-			return a.pipelines.ReadWrite(stm).Update(pipeline, &curPI,
-				func() error { /* no modification, just r+w */ return nil })
-		})
-		if col.IsErrNotFound(err) {
-			log.Warnf("%q polling conflicted with delete (not found)", pipeline)
-		} else if err != nil {
-			// No sensible error recovery here, just keep polling...
-			log.Errorf("could not poll %q: %v", pipeline, err)
-		}
+		m.eventCh <- &pipelineEvent{eventType: writeEv, pipeline: pipeline}
 
-		// move to next pipeline
+		// move to next pipeline (after 2s sleep)
 		return backoff.ErrContinue
-	}, backoff.NewConstantBackOff(2*time.Second),
+	}, backoff.NewConstantBackOff(pollBackoffTime),
 		backoff.NotifyContinue("pollPipelines"),
 	); err != nil {
 		if ctx.Err() == nil {
 			panic("pollPipelines is exiting prematurely which should not happen; restarting pod...")
 		}
 	}
 }
+
+func (m *ppsMaster) pollPipelinePods(pollClient *client.APIClient) {
+	ctx := pollClient.Ctx()
+	if err := backoff.RetryUntilCancel(ctx, func() error {
+		// watchChan will be nil if the Watch call below errors, this means
```

Comment:
seems like we don't need this comment anymore

Reply:
:+1: Removed